cmd/serve.go

```diff
@@ -79,32 +79,45 @@ func elementsHandler(config httpConfig, w http.ResponseWriter, r *http.Request)
 	path = strings.Replace(path, "/", ".", -1)
 	debug.Printf("Element path requested: %s", path)
 
-	config.EConfig.Path = path
-	elements := e.Elements{
-		Config: config.EConfig,
+  format := r.FormValue("format")
+	if format == "" {
+		format = config.OConfig.Format
 	}
 
+	config.OConfig.Format = format
+
 	output := o.Output{
 		Config: config.OConfig,
 	}
 
-	collectedElements, err := elements.Get()
-	if err != nil {
-		return &httpError{err, "Error processing elements", 500}
+	config.EConfig.Path = path
+	elements := e.Elements{
+		Config: config.EConfig,
 	}
 
-	formattedOutput, err := output.Generate(collectedElements)
+	collectedElements, err := elements.Get()
 	if err != nil {
-		return &httpError{err, "Error processing elements", 500}
+		return &httpError{err, "Error collecting elements", 500}
 	}
```

Comment:
As an aside, because the suggested change I made might seem confusing, here we have:

```go
collectedElements, err := elements.Get()
if err != nil {
  return &httpError{err, "Error collecting elements", 500}
}
```

The reason the following form is not done:

```go
collectedElements, err := elements.Get(); err != nil {
  return &httpError{err, "Error collecting elements", 500}
}
```

is because `collectedElements` will then be scoped to the `if` block only, preventing it from being used in line 103 below. In line 103, Go will complain that `collectedElements` does not exist as a variable.